src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor.cs

```diff
@@ -0,0 +1,543 @@
+            Buffer2D<ComplexVector4> targetValues,
+            Buffer2D<ComplexVector4> sourceValues,
+            Rectangle sourceRectangle,
+            Complex64[] kernel,
+            Configuration configuration)
+        {
+            int startY = sourceRectangle.Y;
+            int endY = sourceRectangle.Bottom;
+            int startX = sourceRectangle.X;
+            int endX = sourceRectangle.Right;
+            int maxY = endY - 1;
+            int maxX = endX - 1;
+
+            var workingRectangle = Rectangle.FromLTRB(startX, startY, endX, endY);
+            int width = workingRectangle.Width;
+
+            ParallelHelper.IterateRows(
+                workingRectangle,
+                configuration,
+                rows =>
+                {
+                    for (int y = rows.Min; y < rows.Max; y++)
+                    {
+                        Span<ComplexVector4> targetRowSpan = targetValues.GetRowSpan(y).Slice(startX);
+
+                        for (int x = 0; x < width; x++)
+                        {
+                            DenseMatrixUtils.Convolve1D(kernel, sourceValues, targetRowSpan, y, x, startY, maxY, startX, maxX);
+                        }
+                    }
+                });
+        }
+
+        /// <summary>
+        /// Applies the gamma correction/highlight to the input pixel buffer.
+        /// </summary>
+        /// <param name="targetPixels">The target pixel buffer to adjust.</param>
+        /// <param name="sourceRectangle">
+        /// The <see cref="Rectangle"/> structure that specifies the portion of the image object to draw.
+        /// </param>
+        /// <param name="configuration">The <see cref="Configuration"/></param>
+        private void ApplyGammaExposure(
+            Buffer2D<TPixel> targetPixels,
+            Rectangle sourceRectangle,
+            Configuration configuration)
+        {
+            int startY = sourceRectangle.Y;
+            int endY = sourceRectangle.Bottom;
+            int startX = sourceRectangle.X;
+            int endX = sourceRectangle.Right;
+
+            var workingRectangle = Rectangle.FromLTRB(startX, startY, endX, endY);
+            int width = workingRectangle.Width;
+            float gamma = this.Gamma;
+
+            ParallelHelper.IterateRowsWithTempBuffer<Vector4>(
+                workingRectangle,
+                configuration,
+                (rows, vectorBuffer) =>
+                    {
+                        Span<Vector4> vectorSpan = vectorBuffer.Span;
+                        int length = vectorSpan.Length;
+
+                        for (int y = rows.Min; y < rows.Max; y++)
+                        {
+                            Span<TPixel> targetRowSpan = targetPixels.GetRowSpan(y).Slice(startX);
+                            PixelOperations<TPixel>.Instance.ToVector4(configuration, targetRowSpan.Slice(0, length), vectorSpan);
+                            Vector4Utils.Premultiply(vectorSpan);
```

Comment:
Instead of explicit premultiplication, you should pass `PixelConversionModifiers.Premultiply` to the methods `.ToVector4(...)` and `.FromVector4(...)`.

Reply:
Fixed in https://github.com/SixLabors/ImageSharp/pull/842/commits/fe6c5026ac62b2971a74845263f179bd4ae06efe.

Reply:
Actually, I think I'll need to leave out the modifier in this `FromVector4` call, as the processor is expecting premultiplied data to work on.